### ISSUE TITLE
Merge PR through github API

### ISSUE THREAD

**Issue by tobiasdiez** [sagemath/sage-release-management#2]
Currently, https://github.com/sagemath/sage-release-management/blob/a246c8667ee8ee2f5723476d18d0ff82b5fd153a/git_sage/repo/release_merge.py#L78-L90 only creates a commit but not (directly) merges the PR. Here is the API for it: https://docs.github.com/en/rest/pulls/pulls?apiVersion=2022-11-28#merge-a-pull-request

**Comment by vbraun** [comment]
Yes because I currently run the buildbot with the merge commit and only push it to develop if the checks pass. A bit like a merge queue.

**Comment by tobiasdiez** [comment]
At https://github.com/sagemath/sage/pull/35062, I'm proposing to activate github's new merge queue feature. What would be needed to make this equivalent to your current setup using the buildbot?

**Comment by vbraun** [comment]
I haven't used it but the real task is to set up CI that adequately covers the supported platforms, not turning on the merge queue feature.

**Comment by mkoeppe** [comment]
What platforms do you test, Volker?

**Comment by vbraun** [comment]
I'm apparently building some platform-independent parts that aren't covered by the current ci, e.g. pdf docs

The builders are here: http://build.sagemath.org/#/builders

**Comment by mkoeppe** [comment]
On that page, where does one see what system packages are installed, and how you configure the build?

**Comment by mkoeppe** [comment]
Yes, we should add a PDF doc builder to the CI that runs for every ticket.

**Comment by vbraun** [comment]
The buildbot workers are all minimal installations, the worker configuration is in https://github.com/sagemath/buildbot-sage/blob/master/sage_bot/workers.py

**Comment by mkoeppe** [comment]
Thanks, I'll take a look.

By the way, a PDF builder is added in https://github.com/sagemath/sage/pull/35169
